### PR TITLE
`Development`: Upgrade from tslint to eslint

### PR DESCRIPTION
**Pull request by ge65cer**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).

### Motivation and Context

TSLint is end-of-life since 2020, we still used it as a [dependency](https://github.com/ls1intum/Artemis/pull/5947/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L159).
See  also: https://github.com/palantir/tslint/issues/4534

### Description

Ran the upgrade steps and made sure that all linting errors are fixed or temporarily ignored.

### Steps for Testing

Make sure the code changes do not change the functionality and everything still works correctly.

#### Code Review
- [ ] Review 1
- [ ] Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,213 +1,214 @@
...
+    "@types/showdown": "2.0.0",
+    "@types/smoothscroll-polyfill": "0.3.1",
+    "@types/sockjs-client": "1.5.1",
+    "@types/uuid": "9.0.0",
+    "@typescript-eslint/eslint-plugin": "5.44.0",
+    "@typescript-eslint/eslint-plugin-tslint": "5.45.1",
+    "@typescript-eslint/parser": "5.44.0",
+    "browser-sync": "2.27.10",
+    "browser-sync-webpack-plugin": "2.3.0",
+    "eslint": "^8.28.0",
```
Comment (krusche) [inline package.json:142]: avoid ^ please

**Review by krusche — CHANGES_REQUESTED** [review]
can you please revert the changes in package.json and package-lock.json related to pure whitespace ? we should avoid this

**Comment by krusche** [comment]
why are there still so many changes in `package-lock.json`?

**Comment by ge65cer** [comment]
@krusche I might have had a different node version, now used node v18.

**Comment by ge65cer** [comment]
Resolved merge conflicts.

**Comment by krusche** [comment]
client-style seems to fail, we should fix that before we merge this PR

**Comment by krusche** [comment]
I run `npm install` once more, now the number of changes is more reasonable

**Comment by b-fein** [comment]
According to the log the CI fails as in GitHub Actions config Node 16 is still used instead of 18 as required by the `package.json`. Should be fixed with #5987.

**Comment by ge65cer** [comment]
Waiting for #5987

**Comment by ge65cer** [comment]
All green now. ✅

**Comment by krusche** [comment]
@ge65cer does this mean we can merge the PR and finally get rid of TSLint? Or should we ask some developers to test / review the changes?

**Comment by ge65cer** [comment]
@krusche From my perspective it can be merged.

**Review by krusche — APPROVED** [review]

**Pull request merged**